### PR TITLE
PMM-7358 New endpoint for Managing components.

### PR DESCRIPTION
**Pull request by BupycHuk**
I have two ideas to implement. 

1. First one is this one.
2. The second one is `ChangeKubernetesCluster` endpoint and field for each component.

What do you think which one is better to keep?

### CODE REVIEW

**Comment by jprukner** [comment]
@BupycHuk I think this one is better.

**Review by jprukner — APPROVED** [review]

**Review by tiagomotasantos — APPROVED** [review]
This one seems best to me. There's no benefit in putting everything in the same request, we always have to separate it in some way because each operator has different components and they can also have components with the same name (e.g. backup).

File: api/managementpb/dbaas/components.proto [hunk 1]
```diff
@@ -58,6 +58,32 @@ message GetPXCComponentsResponse {
...
+message SetPSMDBComponentsRequest {
+  // Kubernetes cluster name.
+  string kubernetes_cluster_name = 1;
+  SetComponent mongod = 2;
+  SetComponent backup = 3;
+}
+
+message SetPSMDBComponentsResponse {}
+
+message SetPXCComponentsRequest {
```
Comment (tiagomotasantos) [inline api/managementpb/dbaas/components.proto:76]: Shouldn't we allow to pass pmm here too?

Comment (BupycHuk) [reply api/managementpb/dbaas/components.proto:76]: Good point. But version server contains outdated pmm client. I'll create a ticket for it.

Comment (BupycHuk) [reply api/managementpb/dbaas/components.proto:76]: https://jira.percona.com/browse/PMM-7814

Comment (tiagomotasantos) [reply api/managementpb/dbaas/components.proto:76]: Got it!! 👍

**Review by jprukner — APPROVED** [review]

**Pull request merged**